xarray/core/dataset.py

```diff
@@ -1491,7 +1501,19 @@ def __setitem__(self, key: Hashable, value) -> None:
                 "cannot yet use a dictionary as a key to set Dataset values"
             )
 
-        self.update({key: value})
+        if isinstance(key, list):
```

Comment:
I don't think you can use `Iterable` here because `isinstance(xr.DataArray(), Iterable)` is `True`.